README.md

```diff
@@ -1,2 +1,199 @@
-# IntelliReview
-AI copilot to review github PRs
+# 🤖 PR Code Reviewer with LangChain + GitHub + VectorDB
+
+A  project that automates pull request (PR) code reviews using Large Language Models (LLMs), code embeddings, and GitHub integration. This app fetches PRs from GitHub, retrieves relevant context using a vector database of code embeddings, and generates insightful code review feedback using an LLM.
+
+---
+
+## 🚀 Features
+
+* 🔍 **Automatic PR Fetching** from GitHub repositories
+* 🧠 **Context-aware Code Reviews** using LLMs
+* 🦮 **Code Embeddings with Vector Search** to retrieve relevant code chunks
+* 📚 **LangChain Integration** for prompt orchestration
+* ⚡ **Top-k Semantic Search** from a vector store to enrich PR diff analysis
+* 📘 **Review Summaries** and **inline suggestions**
+
+---
+
+## 🧱 Architecture
+
+```
+          GitHub PR API
+                ↓
+         PR Metadata + Diff
+                ↓
+          Vector Store Query
+     (Top-k most relevant code chunks)
+                ↓
+         ┌────────────────────────┐
+         │   LangChain LLM    │
+         │  Prompt Template   │
+         └────────────────────────┘
+                ↓
+        Structured Review Feedback
+                ↓
+           Markdown Output / PR Comment
+```
+
+---
+
+## 🛠️ Tech Stack
+
+* **GitHub API** – Pull Request integration
+* **LangChain** – Prompting and LLM chain management
+* **OpenAI / Anthropic / LLM of choice** – For generating the review
+* **FAISS & Pinecone** – Vector DB backend
+* **Python** – Primary language
+
+---
+
+## 🏦 Project Structure
+
+```
+.
+├── .env                          # Environment variables
+├── main.py                       # Entry point
+├── github_client.py              # GitHub API wrapper
+├── vector_store.py               # Code embedding and retrieval logic
+├── llm_review.py                 # LangChain code review pipeline
+├── review_templates.py           # Prompt engineering for reviews
+├── utils.py                      # Helper functions
+├── README.md
+└── requirements.txt
+```
+
+---
+
+## ⚙️ Setup Instructions
+
+### 1. Clone the Repo
+
+```bash
+git clone https://github.com/anmolnehru/IntelliReview.git
+cd IntelliReview
+```
+
+### 2. Install Dependencies
+
+```bash
+python -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
+```
+
+### 3. Configure Environment Variables
+
+Create a `.env` file with the following:
+
+```env
+GITHUB_TOKEN=ghp_XXXXXXXXXXXXXXXXXX
+OPENAI_API_KEY=sk-XXXXXXXXXXXXXXXXXX
+VECTOR_DB_PATH=./vectorstore/index.faiss
+REPO_URL=https://github.com/anmolnehru/IntelliReview
+```
+
+### 4. Run the Application
+
+```bash
+python main.py --pr_number 123
```

Comment:
this is also not consistent with main.py behavior, but it's fine to leave as is for now